cl_dll/hud_jumpspeed.cpp

```diff
@@ -0,0 +1,97 @@
+#include <cmath>
+#include <cstring>
+
+#include "hud.h"
+#include "cl_util.h"
+#include "parsemsg.h"
+#include "hudgl.h"
+
+#define FADE_DURATION_JUMPSPEED 0.7f
+
+int CHudJumpspeed::Init()
+{
+	m_iFlags = HUD_ACTIVE;
+
+	hud_jumpspeed = CVAR_CREATE("hud_jumpspeed", "0", FCVAR_ARCHIVE);
+	hud_jumpspeed_below_cross = CVAR_CREATE("hud_jumpspeed_below_cross", "0", FCVAR_ARCHIVE);
+
+	gHUD.AddHudElem(this);
+	return 0;
+}
+
+int CHudJumpspeed::VidInit()
+{
+	passedTime = FADE_DURATION_JUMPSPEED;
+	return 1;
+}
+
+int CHudJumpspeed::Draw(float flTime)
+{
+	if (hud_jumpspeed->value == 0.0f)
+		return 0;
+
+	int r, g, b;
+	UnpackRGB(r, g, b, gHUD.m_iDefaultHUDColor);
+
+	int y;
+	if (hud_jumpspeed_below_cross->value != 0)
+		y = ScreenHeight / 2 + gHUD.m_iFontHeight / 2 + gHUD.m_iFontHeight;
+	else
+		y = ScreenHeight - gHUD.m_iFontHeight - gHUD.m_iFontHeight / 2 - gHUD.m_iFontHeight;
+
+	static float lastTime = flTime;
```

Comment:
A bit cursed to use `static` here where we are coding a proper HUD class which has member fields. This one belongs in a member field imo.